includes/discovery/entity-physical/eltex-mes23xx.inc.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/*
+ * LibreNMS discovery module for Eltex-MES23xx SFP inventory items
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       https://www.librenms.org
+ *
+ * @copyright  2022 Peca Nesovanovic
+ * @author     Peca Nesovanovic <peca.nesovanovic@sattrakt.com>
+ */
+echo "\nCaching OIDs:";
+
+$entity_array = [];
+echo ' ELTEX-MES23xx';
+$trans = snmpwalk_cache_multi_oid($device, 'eltPhdTransceiverInfoEntry', [], 'ELTEX-MES-PHYSICAL-DESCRIPTION-MIB');
+echo ' entAliasMappingIdentifier';
+$mapping = snmpwalk_cache_multi_oid($device, 'entAliasMappingIdentifier', [], 'ENTITY-MIB:IF-MIB');
+
+function normData($par = null)
+{
+    $tmp = str_replace([':', ' '], '', trim(strtoupper($par)));
+    $ret = preg_match('/^[0-9A-F]+$/', $tmp) ? hex2str($tmp) : $par;
+
+    return $ret;
+}
+
+foreach ($trans as $index => $data) {
+    unset($connectedto);
+    foreach ($mapping as $ekey => $edata) {
+        if ($edata['entAliasMappingIdentifier'] == 'ifIndex.' . $index) {
+            $connectedto = explode('.', $ekey)[0];
+        }
+    }
+    if ($connectedto) {
+        $entity_array[] = [
+            'entPhysicalIndex'        => $index,
+            'entPhysicalDescr'        => $data['eltPhdTransceiverInfoType'],
+            'entPhysicalClass'        => 'sfp-cage',
+            'entPhysicalName'         => strtoupper($data['eltPhdTransceiverInfoConnectorType']),
+            'entPhysicalModelName'    => normData($data['eltPhdTransceiverInfoPartNumber']),
+            'entPhysicalSerialNum'    => $data['eltPhdTransceiverInfoSerialNumber'],
+            'entPhysicalContainedIn'  => $connectedto,
+            'entPhysicalMfgName'      => $data['eltPhdTransceiverInfoVendorName'],
+            'entPhysicalHardwareRev'  => normData($data['eltPhdTransceiverInfoVendorRev']),
+            'entPhysicalIsFRU'        => 'true',
+        ];
+    }
+}
+
+foreach ($entity_array as $entPhysicalIndex => $entry) {
+    $entPhysicalIndex = array_key_exists('entPhysicalIndex', $entry) ? $entry['entPhysicalIndex'] : '';
```

Comment:
for next time, you can simplify it like:
```suggestion
    $entPhysicalIndex = $entry['entPhysicalIndex'] ?? '';
```

Reply:
also, if these fields can be null they should be rather than a empty  string

Reply:
@Jellyfrog 
last time i spoke with @murrant he said there is no need for json, but no problem, uploded one

Reply:
@Jellyfrog 
done